### ISSUE TITLE
Updating matplotlib to 2.0.0 causing problems when plotting: error/warning message "failed to get the current screen resources"

### ISSUE THREAD

**Issue by jdhao** [matplotlib/matplotlib#8058]
### Bug report

**Bug summary**

After updating the matplotlib to the latest version (2.0.0),  an error/warning message occurs
>failed to get the current screen resources
QXcbConnection: XCB error: 170 (Unknown), sequence: 164, resource id: 90, major code: 146 (Unknown),   minor code: 20

**More detailed info**
After the updating, I found a lot of annoying problems.  Following the instrcutions at https://github.com/ContinuumIO/anaconda-issues/issues/1249, I was able to fix the problem.
>This application failed to start because it could not find or load the Qt platform plugin "xcb"
in "".
Available platform plugins are: minimal, offscreen, xcb.
Reinstalling the application may fix this problem.
Aborted

But there is another annoying problem, every time I execute a python script in the shell (for example, `python simple.py`), another error/warning message occured, which is in the bug summary section.

**Code for reproduction**
The **simple.py** script could be 
```
import matplotlib.pyplot as plt
x = [1, 2, 3]
y = [2, 4, 6]
plt.plot(x, y)
plt.show()
```
This picture below shows the error message when executing the script
![qq 20170210164216](https://cloud.githubusercontent.com/assets/16662357/22820148/7d20c6e0-efb0-11e6-9aa1-a258a2eb0c91.png)
If I comment out ` plt.show` or replace it with `plt.savefig('simple.py')` the error message becomes solely
> failed to get the current screen resources

I have search the web, but found no helpful message about how to solve this issue. I am not familiar with the qt and xcb package,  I guess this still has something to do with the qt package and xcb.  

**Matplotlib version**
The matplotlib version now is 2.0.0 and I connect to the server via ssh using putty and xming. The server
is installed with CentOS Linux release 7.2.1511.

**Comment by jenshnielsen** [comment]
I don't think this is a matplotlib issue as such 

The conda version of matplotlib is using QT as the default GUI backend. It looks like QT does not know how to render a window x forwarding. Can you open a simple pyqt5 program such as 


```

"""
ZetCode PyQt5 tutorial 

In this example, we create a simple
window in PyQt5.

author: Jan Bodnar
website: zetcode.com 
last edited: January 2015
"""

import sys
from PyQt5.QtWidgets import QApplication, QWidget


if __name__ == '__main__':
    
    app = QApplication(sys.argv)

    w = QWidget()
    w.resize(250, 150)
    w.move(300, 300)
    w.setWindowTitle('Simple')
    w.show()
    
    sys.exit(app.exec_())
```
from  http://zetcode.com/gui/pyqt5/firstprograms/

If that doesn't work you either need to figure out how to get qt to work over ssh and xforwarding 
or chose a different matplotlib backend. You can try the Tk backend by adding 

```
import matplotlib
matplotlib.use('TKAgg')
```

to the top of your example

**Comment by jenshnielsen** [comment]
If you don't want interactive figures just generate images you can also select a non interactive backend. Add to the top

```
import matplotlib
matplotlib.use('Agg')
```

**Comment by jdhao** [comment]
@jenshnielsen , I tried the above qt code and the same error message occured.

**Comment by jdhao** [comment]
@jenshnielsen , I do a lot of interactive plotting using jupyter-console, so I would prefer an interactive backend. The original matplotlib with Anaconda 2 distribution is version 1.5.1, the qt version is 4.X,  I aslo use xforwarding and they
work smoothly together.  Why can not newer version of them work together?

**Comment by jenshnielsen** [comment]
Given that worked with qt4 and does not work with qt5 this seems like a difference between pyqt4 and pyqt5. 

Both matplotlib 1.5 and 2.0 support both pyqt4 and pyqt5. The only thing that has changed is that conda now packages pyqt5 and not pyqt4. The conda matplotlib package of 2.0 depends on pyqt5 but that is a conda specific choice how that is handled and nothing that we can change in matplotlib. It is perfectly possible to use Matplotlib 2.0 with pyqt4.

**Comment by jenshnielsen** [comment]
I suggest investigating x forwarding with pyqt5 using the simple example i quoted above or using a different backend such as Tk as suggested above.

**Comment by jdhao** [comment]
@jenshnielsen , I will investigate the issue further and switch over to other backend for now. Thanks for your advices.

**Comment by tacaswell** [comment]
The conda-forge build of Matplotilb 2.0 is packaged to use qt4.

This is another issue that ties back to https://github.com/ContinuumIO/anaconda-issues/issues/1068

**Comment by jdhao** [comment]
@tacaswell , I have tried the suggestion proposed there. But if I designate the qt version by using `conda install pyqt=4.11`, several packages including matplotlib in downgraded. The downgraded matplotlib version is 1.5.1, see the following message
>[~]$ conda install pyqt=4.11
Fetching package metadata .........
Solving package specifications: .
>
>Package plan for installation in environment /home/jdhao/anaconda2:
>
>The following packages will be DOWNGRADED due to dependency conflicts:
>
>   anaconda-navigator: 1.4.3-py27_0      --> 1.2.3-py27_0
> matplotlib:         2.0.0-np111py27_0 --> 1.5.1-np111py27_0
>    pyqt:               5.6.0-py27_2      --> 4.11.4-py27_4
>    qt:                 5.6.2-3           --> 4.8.7-4
>    qtconsole:          4.2.1-py27_1      --> 4.2.1-py27_0
>
>Proceed ([y]/n)?

How can i install matplotlib 2.0.0 and pyqt 4.x at the same time?

**Comment by tacaswell** [comment]
you need to include the conda-forge channel `conda install pyqt=4.11 -c conda-forge` which will change the versions on a lot of your packages.  See https://conda-forge.github.io/

You should contact continuum about them providing a build of Matplotlib pinned to qt4, we have no control over that. 

Closing this as either an upstream problem (qt5 not playing nice across x-forwarding) or a downstream packaging problem.

**Comment by jdhao** [comment]
@tacaswell , `conda install pyqt=4.11 -c conda-forge` is not working, the matplotlib is still to be downgraded though. see the picture below
![qq 20170210233808](https://cloud.githubusercontent.com/assets/16662357/22832693/0a7c5a5c-efea-11e6-91ad-098235d75be8.png)

**Comment by tacaswell** [comment]
Try `conda create -n new_env matplotlib=2.0.0 ipython -c conda-forge`.  Creating a new enviroment is safer (won't break your root enviroment with weird dependencies changes) and in general better practice.

**Comment by jdhao** [comment]
@tacaswell . Thank you. `onda create -n new_env matplotlib=2.0.0 ipython -c conda-forge` finally installs matplotlib 2.0.0 with qt 4.x. Then the above error message does not occur. A lot of thanks!

**Comment by mingwandroid** [comment]
@jdhao, can you try the same thing but connecting from something other than Xming? It's unclear to me which package contains this bug at present.

**Comment by jdhao** [comment]
@mingwandroid ,sorry for the late reply. I use the commercial [x-win32](https://www.starnet.com/xwin32/) to replace Xming. I set up two virtual env, the first env with matplotlib 2.0.0 and qt/pyqt 5.x, the second env with matplotlib 2.0.0 and qt/pyqt 4.x.  The error message occurs in the first env but not in the second env.
Apparently it is qt (not x11 forwarding software nor matplotlib) that is causing the problem.

**Comment by mingwandroid** [comment]
@jdhao, do you mean you get the same result from both Xming and x-win32?

**Comment by jdhao** [comment]
@mingwandroid, yes, the same result for both Xming and x-win32.

**Comment by mingwandroid** [comment]
OK thanks.
